File: q2706/q2706.cpp
class Solution
{
public:
    int buyChoco(vector<int> &prices, int money)
    {
        int lowest = INT_MAX;
        int secondLowest = INT_MAX;
        for (const int price : prices)
        {
            if (price < lowest)
            {
                secondLowest = lowest;
                lowest = price;
            }
            else
            {
                secondLowest = min(secondLowest, price);
            }
        }
        return money >= (lowest + secondLowest) ? money - (lowest + secondLowest) : money;
    }
};

File: q2706/q2706.java
class Solution {
    public int buyChoco(int[] prices, int money) {
        int lowest = Integer.MAX_VALUE;
        int secondLowest = Integer.MAX_VALUE;
        for (int price : prices) {
            if (price < lowest) {
                secondLowest = lowest;
                lowest = price;
            } else {
                secondLowest = Math.min(secondLowest, price);
            }
        }
        return money >= (secondLowest + lowest) ? money - (secondLowest + lowest) : money;
    }
}